src/main/java/edu/hm/hafner/grading/FileSystemToolParser.java

```diff
@@ -77,6 +77,37 @@ public Node readNode(final ToolConfiguration tool, final FilteredLog log) {
         }
     }
 
+    @Override
+    public Node readDeltaNode(final ToolConfiguration tool, final FilteredLog log) {
+        var parser = new edu.hm.hafner.coverage.registry.ParserRegistry().get(StringUtils.upperCase(tool.getId()),
+                ProcessingMode.IGNORE_ERRORS);
+
+        var nodes = new ArrayList<Node>();
+        for (Path file : REPORT_FINDER.findDelta(log, getDisplayName(tool), tool.getPattern())) {
+            var factory = new FileReaderFactory(file);
+            try (var reader = factory.create()) {
+                var node = parser.parse(reader, file.toString(), log);
+                log.logInfo("- %s: %s", PATH_UTIL.getRelativePath(file), extractMetric(tool, node));
+                nodes.add(node);
+            }
+            catch (IOException exception) {
+                throw new ParsingException(exception);
+            }
+        }
+
+        if (nodes.isEmpty()) {
+            return createEmptyContainer(tool);
+        }
+        else {
+            var aggregation = Node.merge(nodes);
+            log.logInfo("-> %s Total: %s", getDisplayName(tool), extractMetric(tool, aggregation));
+            // Wrap the node into a container with the specified tool name
+            var containerNode = createEmptyContainer(tool);
+            containerNode.addChild(aggregation);
+            return containerNode;
+        }
+    }
+
```

Comment:
Can't we use the same code as in `readNode` here? The only thing that changes is the directory of the artefacts.